src/main/java/at/hannibal2/skyhanni/features/misc/InWaterDisplay.kt

```diff
@@ -0,0 +1,30 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import net.minecraft.client.Minecraft
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class InWaterDisplay {
+
+    private val config get() = SkyHanniMod.feature.misc.stranded
+    private var playerInWater = false
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        playerInWater = Minecraft.getMinecraft().thePlayer.isInWater
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+
+        val text = "In Water: " + if (playerInWater) "§aTrue" else "§cFalse"
```

Comment:
Why don't you just directly check the player inwater here

Reply:
huge oversight by me, sorry i'm new to this

Reply:
No worries

Reply:
i don't think there's a need to check every frame instead of every tick

Reply:
player inwater is already a boolean that is updated every tick

Reply:
oh in that case it's fine, i had no idea how the check worked